File: Leetcode/27-Remove-Element/RemoveElement.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        if (nums.empty()) {
            return 0;
        }
        int len = 0;
        for (auto i=nums.begin(), j = nums.begin(); j!= nums.end(); j++) {
            if (*j!=val) {
                *i = *j;
                i++;
                len++;
            }
        }
        return len;
    }
};

File: Leetcode/27-Remove-Element/RemoveElement.py
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        if nums == None or len(nums) == 0:
            return 0
        i=0
        j=0
        length = len(nums)
        while j < length :
            if nums[j]!=val :
                nums[i] = nums[j]
                i+=1
            j+=1
        return i

